All files

Lukease opened the pull request.

Commit:
initial commit

Commit:
Finish Project

jpudysz commented on cats/style.sass:
```diff
@@ -0,0 +1,104 @@
+.hide
+  transform: rotateY(90deg)
+
+//.show
```
remove comment

jpudysz commented on cats/main.js:
```diff
@@ -0,0 +1,139 @@
+    return arena
+}
+
+let number = 0
```
number of what?

jpudysz commented on cats/main.js:
```diff
@@ -0,0 +1,139 @@
+}
+
+const arena = createArena()
+let scoreBox = createScoreBox()
```
why let?

jpudysz commented on cats/main.js:
```diff
@@ -0,0 +1,139 @@
+let clickedIndex = []
+
+cardsArray.forEach((object) => {
+    const newCard = document.createElement('div')
+
+    newCard.classList.add('card')
+    arena.appendChild(newCard)
+
+    const cardBack = document.createElement('div')
+
+    cardBack.classList.add('card__back')
+    newCard.appendChild(cardBack)
```
you can move it to separate function like `createCard`

jpudysz commented on cats/main.js:
```diff
@@ -0,0 +1,139 @@
+    cardBack.classList.add('card__back')
+    newCard.appendChild(cardBack)
+
+    const click = (event) => {
```
arrow function with one argument should be without `()`

jpudysz commented on cats/main.js:
```diff
@@ -0,0 +1,139 @@
+            event.target.classList.add('hide')
+            clickLimit--
+            event.target.nextElementSibling.classList.add('show')
+            clickedList.push(event.target.nextElementSibling.src)
```
use concat, push is banned

jpudysz commented on cats/main.js:
```diff
@@ -0,0 +1,139 @@
+    const randomNumber = Math.floor(Math.random() * filtered.length)
+    let randomNumberValue = filtered[randomNumber]
+
+    cardsArray[randomNumberValue] = 'cat'
```
hmm?

Commit:
remove comment,
change name from number to scoreNumber,
change let to const,
create new function createCard(),
delete () from click function,
use concat on clickedList.

jpudysz commented on cats/main.js:
```diff
@@ -36,22 +36,51 @@ const createScoreBox = () => {
-let clickedList = []
-let clickLimit = 2
-let clickedIndex = []
+    if (clickLimit === 0 && clickedList[0] === clickedList[1]) {
```
try not to reference element in array by [0], use destructuring

jpudysz commented on cats/main.js:
```diff
@@ -36,22 +36,51 @@ const createScoreBox = () => {
+            event.target.removeEventListener('click', click)
 
-cardsArray.forEach((object) => {
+            let points = document.querySelector('.score__value')
```
why let?

jpudysz commented on cats/main.js:
```diff
@@ -36,22 +36,51 @@ const createScoreBox = () => {
+        }, 1000)
+    }
+
+    if (clickLimit === 0 && clickedList[0] !== clickedList[1]) {
```
aa

jpudysz commented on cats/main.js:
```diff
@@ -62,46 +91,22 @@ cardsArray.forEach((object) => {
+const arena = createArena()
+const scoreBox = createScoreBox()
+
+let cardsArray = Array.from(new Array(20)).map(() => '')
```
reconsider changing it to const

jpudysz commented on cats/main.js:
```diff
@@ -62,46 +91,22 @@ cardsArray.forEach((object) => {
+
+let cardsArray = Array.from(new Array(20)).map(() => '')
+const imagesLocation = './cats/'
+let imagesArray = Array.from(new Array(10))
```
you have 2 arrays for same thing, please use only one array

jpudysz commented on cats/main.js:
```diff
@@ -62,46 +91,22 @@ cardsArray.forEach((object) => {
+let clickLimit = 2
+let clickedIndex = []
+
+cardsArray.forEach((object) => {
```
you can use shortcut:
```
cardsArray.forEach(createCards)
```

Commit:
used destructuring,
delete cardsArray,
create new function addHiddenImages(),
create new array sortedArrayOfImages,
use forEach on sortedArrayOfImages.

jpudysz commented on cats/main.js:
```diff
@@ -50,21 +50,28 @@ const click = event => {
 
-    if (clickLimit === 0 && clickedList[0] === clickedList[1]) {
+    const [firstCard, secondCard] = clickedList
+    if (clickLimit === 0 && firstCard === secondCard) {
```
enter before if

jpudysz commented on cats/main.js:
```diff
@@ -95,50 +97,32 @@ const createCards = () =>{
-    newCard[randomNumberValue].appendChild(image)
+const addHiddenImages = () =>{
+    sortedArrayOfImages = imagesArray.concat(imagesArray)
+    sortedArrayOfImages.sort((a,b)=>0.5-Math.random())
```
space after `,` and between `-`

jpudysz replied:
you can use chainging:
```
return imagesArray
    .concat(imagesArray)
    .sort((a, b)=>0.5 - Math.random())
```

Lukease replied:
it doesn't work

jpudysz commented on cats/main.js:
```diff
@@ -95,50 +97,32 @@ const createCards = () =>{
-        .map((field, index) => field ? null : index)
-        .filter(item => item !== null)
-    const secondRandomNumber = Math.floor(Math.random() * filteredSecond.length)
+sortedArrayOfImages.forEach((object,index)=>{
```
space between arguments

Commit:
add spaces.
